From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill <iyaargg@gmail.com>
Date: Sun, 30 Jun 2024 08:40:15 +0500
Subject: [PATCH] Add my name to startup message.


diff --git a/src/main/java/net/minecraft/server/DedicatedServer.java b/src/main/java/net/minecraft/server/DedicatedServer.java
index 2519d4ca1bb09d76caa2373164625aa6b692176e..8371b5dcc6b1088543a3baffa6e72a885405edff 100644
--- a/src/main/java/net/minecraft/server/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/DedicatedServer.java
@@ -128,6 +128,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
         thread.setDaemon(true);
         thread.start();
         DedicatedServer.LOGGER.info("Starting minecraft server version 1.8.8");
+        DedicatedServer.LOGGER.info(">> Modified by Rejomy");
         if (Runtime.getRuntime().maxMemory() / 1024L / 1024L < 512L) {
             DedicatedServer.LOGGER.warn("To start the server with more ram, launch it as \"java -Xmx1024M -Xms1024M -jar minecraft_server.jar\"");
         }
