From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill <iyaargg@gmail.com>
Date: Sat, 6 Jul 2024 12:02:28 +0500
Subject: [PATCH] Supress Invalid Book Exceptions


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index e21ce6f8434a7eba8d1137f46e35493b4863736d..8261a8cdc82e951d53c06659ce0ac8f65c65e12e 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -2144,7 +2144,7 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                     return;
                 }
             } catch (Exception exception1) {
-                PlayerConnection.c.error("Couldn\'t sign book", exception1);
+                // FlamePaper - Supress Invalid Book Exceptions
                 this.disconnect("Invalid book data!"); // CraftBukkit
                 return;
             } finally {
