From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill <iyaargg@gmail.com>
Date: Sat, 6 Jul 2024 12:54:11 +0500
Subject: [PATCH] Do not load chunk for light check


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 2dd03cc25b30ed93c1cdcaa7802cb9ecdcd9c9f7..7a713068d1cba162bbbad5c29df01741403c186c 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -672,6 +672,10 @@ public abstract class World implements IBlockAccess {
                 blockposition = new BlockPosition(blockposition.getX(), 255, blockposition.getZ());
             }
 
+            // FlamePaper start - Do not load chunks for light checks
+            if (!this.isLoaded(blockposition)) return 0;
+            // FlamePaper end - Do not load chunks for light checks
+
             return this.getChunkAtWorldCoords(blockposition).a(blockposition, 0);
         }
     }
@@ -749,6 +753,8 @@ public abstract class World implements IBlockAccess {
                     blockposition = new BlockPosition(blockposition.getX(), 255, blockposition.getZ());
                 }
 
+                if (!this.isLoaded(blockposition)) return 0; // Paper
+
                 Chunk chunk = this.getChunkAtWorldCoords(blockposition);
 
                 return chunk.a(blockposition, this.I);
